### PR TITLE
Beef up examples

### PR DESCRIPTION
**Pull request by edgarmueller**

### CODE REVIEW

**Comment by edgarmueller** [comment]
This also closes #190

File: components/renderers/controls/array/array-renderer.ts [hunk 1]
```diff
@@ -122,7 +123,7 @@ class ArrayRenderer implements JSONForms.IRenderer {
                 containerDescription['submitControls'] = this.createControlsForSubmit(items, schemaPath, submitElement, services);
                 if (data == undefined) {
                     containerDescription["instance"][containerDescription['path']] = [];
-                    data = [];
+                    data = containerDescription["instance"][containerDescription['path']];
```
Comment (sdirix) [inline components/renderers/controls/array/array-renderer.ts]: I liked the previous line better

Comment (edgarmueller) [inline components/renderers/controls/array/array-renderer.ts]: Yeah, but it doesn't work. `data` need to point to the actual reference and not to a anonymous array.

**Comment by sdirix** [comment]
First things first: This looks GREAT!

Minor nitpicks:
 * ```#editor```: 

image
 * ```#custom```, ```#defaultui```, ```#defaultschema```:

image
 * ```#placeholder```: 

image

Good work! I like the look of the new examples!

**Comment by edgarmueller** [comment]
Incoming, btw: does the editor example work for you?

**Comment by sdirix** [comment]
No it does not. Also there are validation problems in the ```#local``` example. When the form is initially rendered the ```Age``` is not validated. Not before I change an unrelated property the ```Required``` finally shows up.
image
image
I assume these problems do not originate from this change? That is why I did not mention them.

File: components/renderers/controls/array/array-renderer.ts (hunk 1 above)
Comment (sdirix) [inline components/renderers/controls/array/array-renderer.ts]: Ah good to know! :+1:

**Comment by edgarmueller** [comment]
This is intended, although I admit, that this is might actually be a bad example. The age field and "that unrelated property" both are contained in a sub-object, which does not exists initially, hence validation does not complain. Only when you change the unrelated property the object will be created and the validation is triggered. 
Could you create another issue for fixing the editor example and continue with this PR?
Thank you!

**Comment by sdirix** [comment]
Great work ;) :+1:

**Comment by edgarmueller** [comment]
Fixed via #188

**Pull request merged**